site/docs/tutorials/snake/drawing-the-snake.md

```diff
@@ -722,6 +782,132 @@ This changes the color back and adds the darker green as its outline.
 
 </Page>
 
+<Page value="wat">
+
+But where is the head? You can pick a side. Either position `[0]` or position `[body_length - 1]`.
+
+I think it's easier to pick `[0]`.
+
+Since the body is drawn, head is not much of a problem. Simply use the `rect` function again. But use a specific part instead:
+
+```wasm
+(call $rect
+  (i32.mul (i32.load (i32.const 0x19ac)) (i32.const 8))
+  (i32.mul (i32.load (i32.const 0x19b0)) (i32.const 8))
+  (i32.const 8)
+  (i32.const 8))
+```
+
+The draw function should now look like this:
+
+```wasm
+(func $snake-draw
+  (local $offset i32)      ;; offset is initialized to 0
+  (local $offset-end i32)
+
+  ;; offset-end = body_length * 8
+  (local.set $offset-end
+    (i32.mul
+      (i32.load (i32.const 0x19a8))  ;; body_length
+      (i32.const 8)))
+
+  ;; loop over all points in the body
+  (loop $loop
+    ;; rect(part.x * 8, part.y * 8, 8, 8)
+    ;; 
+    ;; Note that the array starts at 0x19a4, so the first x-coordinate starts
+    ;; at 0x19a4 and the first y-coordinate starts at 0x19a8. We can bake these
+    ;; offsets directly into the instruction to reduce the code size.
+    (call $rect
+      (i32.mul (i32.load offset=0x19ac (local.get $offset)) (i32.const 8))
+      (i32.mul (i32.load offset=0x19b0 (local.get $offset)) (i32.const 8))
+      (i32.const 8)
+      (i32.const 8))
+
+    ;; Add 8 to offset, and loop if offset < offset-end.
+    (br_if $loop
+      (i32.lt_u
+        (local.tee $offset (i32.add (local.get $offset) (i32.const 8)))
+        (local.get $offset-end)))
+  )
+
+  ;; Draw the head.
+  (call $rect
+    (i32.mul (i32.load (i32.const 0x19ac)) (i32.const 8))
+    (i32.mul (i32.load (i32.const 0x19b0)) (i32.const 8))
+    (i32.const 8)
+    (i32.const 8))
+)
+```
+
+Notice the difference? Me neither.
+
+The head should stand out a little. For this, you can use a different color (remember that `DRAW_COLORS` is located at address 0x14):
```

Comment:
Lack of named constants is definitely a bit of a bummer. Maybe m4 + $LANG isn't such a bad idea after all ;)

Reply:
Honestly, I was thinking of doing this for my own WAT projects. :smiley:

Reply:
Hm, you can actually use immutable globals for this, at the cost of extra file size. I'll make that change too.

Reply:
Ah that makes sense. It seems like `wasm-opt` ought to be able to inline those too?

Reply:
I think so, though I'm not as sure about this optimization.